pkg/controller/dual-pods/controller.go

```diff
@@ -308,27 +308,42 @@ type infSvrItem struct {
 	RequesterName string
 }
 
-// careAbout returns infSvrItem, podIsRequester, have.
-// Returns have=true for both requesters and bound providers,
-// have=false for unbound providers and other Pods.
-func careAbout(pod *corev1.Pod) (infSvrItem, bool, bool) {
+type infSvrItemType string
+
+const (
+	// infSvrItemRequester is for a server-requesting Pod.
+	infSvrItemRequester infSvrItemType = "requester"
+	// infSvrItemBoundDirectProvider is for a server-providing Pod that
+	// is 'direct' (i.e. not launcher-based), and bound to a server-requesting Pod.
+	infSvrItemBoundDirectProvider infSvrItemType = "bound_direct_provider"
+	// infSvrItemLauncherBasedProvider is for a server-providing Pod that is launcher-based.
+	infSvrItemLauncherBasedProvider infSvrItemType = "launcher_based_provider"
+	// infSvrItemDontCare is not a real infSvrItemType but only a placeholder
+	// saying the corresponding infSvrItem is not relevant to the controller.
+	infSvrItemDontCare infSvrItemType = "dont_care"
+)
+
+// careAbout returns an infSvrItem item and a bool owned.
+// Returns owned=true for server-requesting Pods, bound direct server-providing Pods, and launcher-based server-providing Pods;
+// returns owned=false for unbound direct providers and other Pods.
+func careAbout(pod *corev1.Pod) (item infSvrItem, it infSvrItemType, owned bool) {
```

Comment:
This could be simplified by removing `owned bool`, because `owned == (it != infSvrItemDontCare)`.